crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -581,14 +581,15 @@ impl<F: Send + Clone> GetTracker<F, PaymentData<F>, api::PaymentsRequest> for Pa
                 .payment_method_data
                 .clone()
                 .map(|x| x.into_inner().expose())
-                .and_then(|v| match serde_json::from_value::<PaymentMethodsData>(v) {
-                    Ok(data) => Some(data),
-                    Err(err) => {
-                        router_env::logger::info!(
-                            "PaymentMethodsData deserialization failed : {err}"
-                        );
-                        None
-                    }
+                .and_then(|v| {
+                    v.parse_value("PaymentMethodsData")
+                        .map_err(|err| {
```

Comment:
Can use `inspect_err` also instead of `map_err` here (and then can skip the error being returned from the closure).

Optional change, by the way.